In [1]:
# def filtering_text(input_file_path, output_file_path):
#     import re  # Import re here

#     def cleanTxt(text):
#         text = re.sub('…', '', text)
#         text = re.sub('&amp;', '', text)
#         text = re.sub('@[A-Za-z0-9]+', '', text)  # Removing @mentions
#         text = re.sub('#', '', text)  # Removing '#' hash tag
#         text = re.sub('|', '', text)  # Removing '|' sign
#         text = re.sub('-', '', text)  # Removing '-' sign
#         text = re.sub('_', '', text)  # Removing '_' sign
#         text = re.sub('https?:\/\/\S+', '', text)  # Removing hyperlink
#         text = re.sub(' +', ' ', text)  # Removing double spaces
#         text = re.sub(' {2,}', ' ', text)  # Removing two or more spaces and replacing with a single space
#         return text

#     # Read the lines from the file and remove duplicates
#     with open(input_file_path, 'r') as file:
#         lines = file.readlines()
#         unique_lines = []

#         for line in lines:
#             if line not in unique_lines:
#                 unique_lines.append(line)

#     # Clean the text
#     cleaned_lines = [cleanTxt(line) for line in unique_lines]

#     # Write the cleaned lines back to the file
#     with open(output_file_path, 'w') as file:
#         file.writelines(cleaned_lines)

# # Example usage:
# input_file_path = "input_file.txt"  # Replace with the actual input file path
# output_file_path = "result_file_01.txt"  # Replace with the desired output file path

# filtering_text(input_file_path, output_file_path)


In [2]:
# def filtering_text(input_file_path, output_file_path):
#     import re  # Import re here

#     # Read the lines from the file and remove duplicates
#     with open(input_file_path, 'r') as file:
#         lines = file.readlines()
#         unique_lines = []

#         for line in lines:
#             if line not in unique_lines:
#                 unique_lines.append(line)

#     # Clean the text
#     cleaned_lines = []

#     for line in unique_lines:
#         line = re.sub('…', '', line)
#         line = re.sub('&amp;', '', line)
#         line = re.sub('@[A-Za-z0-9]+', '', line)  # Removing @mentions
#         line = re.sub('#', '', line)  # Removing '#' hash tag
#         line = re.sub('|', '', line)  # Removing '|' sign
#         line = re.sub('-', '', line)  # Removing '-' sign
#         line = re.sub('_', '', line)  # Removing '_' sign
#         line = re.sub('https?:\/\/\S+', '', line)  # Removing hyperlink
#         line = re.sub(' +', ' ', line)  # Removing double spaces
#         line = re.sub(' {2,}', ' ', line)  # Removing two or more spaces and replacing with a single space
#         cleaned_lines.append(line)

#     # Write the cleaned lines back to the file
#     with open(output_file_path, 'w') as file:
#         file.writelines(cleaned_lines)

# # Example usage:
# input_file_path = "input_file.txt"  # Replace with the actual input file path
# output_file_path = "result_file_01.txt"  # Replace with the desired output file path

# filtering_text(input_file_path, output_file_path)


In [3]:
# def filtering_text(input_text):
#     import re  # Import re here

#     # Split the input text into lines
#     lines = input_text.splitlines()
#     unique_lines = []

#     # Remove duplicates
#     for line in lines:
#         if line not in unique_lines:
#             unique_lines.append(line)

#     # Clean the text
#     cleaned_lines = []

#     for line in unique_lines:
#         line = re.sub('…', '', line)
#         line = re.sub('&amp;', '', line)
#         line = re.sub('@[A-Za-z0-9]+', '', line)  # Removing @mentions
#         line = re.sub('#', '', line)  # Removing '#' hash tag
#         line = re.sub('|', '', line)  # Removing '|' sign
#         line = re.sub('-', '', line)  # Removing '-' sign
#         line = re.sub('_', '', line)  # Removing '_' sign
#         line = re.sub('https?:\/\/\S+', '', line)  # Removing hyperlink
#         line = re.sub(' +', ' ', line)  # Removing double spaces
#         line = re.sub(' {2,}', ' ', line)  # Removing two or more spaces and replacing with a single space
#         cleaned_lines.append(line)

#     # Join the cleaned lines back into a string
#     cleaned_text = '\n'.join(cleaned_lines)

#     return cleaned_text

# # Example usage:
# input_text = "Your large input string here..."
# output_text = filtering_text(input_text)

# # Print or use the cleaned text as needed
# print(output_text)


In [4]:
import re

def cleanTxt(text):
    text = re.sub('…', '', text)
    text = re.sub('&amp;', '', text)
    text = re.sub('@[A-Za-z0-9]+', '', text)  # Removing @mentions
    text = re.sub('#', '', text)  # Removing '#' hash tag
    text = re.sub('|', '', text)  # Removing '|' sign
    text = re.sub('-', '', text)  # Removing '-' sign
    text = re.sub('_', '', text)  # Removing '_' sign
    text = re.sub('https?:\/\/\S+', '', text)  # Removing hyperlink
    text = re.sub(' +', ' ', text)  # Removing double spaces
    text = re.sub(' {2,}', ' ', text)  # Removing two or more spaces and replacing with a single space

    return text


def filtering_text(input_file_path, output_file_path):
    # Read the lines from the file and remove duplicates
    with open(input_file_path, 'r') as file:
        lines = file.readlines()
        unique_lines = []

        for line in lines:
            if line not in unique_lines:
                unique_lines.append(line)

    # Clean the text
    cleaned_lines = [cleanTxt(line) for line in unique_lines]

    # Write the cleaned lines back to the file
    with open(output_file_path, 'w') as file:
        file.writelines(cleaned_lines)



# Example usage:
input_file_path = "output_result.txt"  # Replace with the actual input file path
output_file_path = "filterd_file.txt"  # Replace with the desired output file path

filtering_text(input_file_path, output_file_path)

# with-out filtring

In [5]:
with open('output_result.txt', 'r') as file:
    input_text_data_lines =  file.readlines()

In [19]:
for index,line in enumerate(input_text_data_lines, start=1):
    print(f"======================================= #Line No {index} # ================================================")
    print(line)

======================================= #Line No 1 # ================================================
History and Physical Examination:

======================================= #Line No 2 # ================================================


======================================= #Line No 3 # ================================================
HISTORY:

======================================= #Line No 4 # ================================================
Presenting Complaints / Reasons: 

======================================= #Line No 5 # ================================================
   T-cell Lymphoblastic Lymphoma

======================================= #Line No 6 # ================================================
   Fever,Recurrent hypoglycaemia

======================================= #Line No 7 # ================================================
   Respiratory distress and loose stools

======================================= #Line No 8 # =========================================

In [7]:
with open('output_result.txt', 'r') as file:
    input_text_data =  file.read()

In [16]:
input_text_data

"History and Physical Examination:\n\nHISTORY:\nPresenting Complaints / Reasons: \n   T-cell Lymphoblastic Lymphoma\n   Fever,Recurrent hypoglycaemia\n   Respiratory distress and loose stools\n\nHistory Of Present illness:\n   T cell lymphoblastic lymphoma    \n   Induction day 16\n   Fever came in emergency with complain of fever for\n   1 day without rigors and chills , associated with\n   cough non productive .Has respiratory distress and\n    recurrent hypoglycaemia\n   Developed loose stools since morning multiple\n   episodes as per attendant\n   persistent Moaning and has weak peripheral\n   pulses.CRT >2 seconds.  \n\nPast Medical History: \n   chemo related multiple admissions\n\nSocioeconomic History:\nMarital Status:\n   minor\n\nFather's profession / Monthly income:\n   dependant\n\nAny drugs of abuse/addictions:\n   none\n\nOthers:\n   none\n\nFamily History: \n   insignificant\n\nDevelopmental History:\n   as per age\n\nVaccination History:\n   as per epi\nREVIEW OF SYSTE

###### Totel Length of Input Text Data

In [8]:
len(input_text_data)

69570

###### Totel Length of Input Text Data Courpus

In [9]:
input_text_data_courpus= input_text_data.split(" ")
len(input_text_data_courpus)

10434

# After filtring

In [10]:
with open('filterd_file.txt', 'r') as file:
    filterd_text_data_lines =  file.readlines()

In [20]:
for index,line in enumerate(filterd_text_data_lines, start=1):
    print(f"======================================= #Line No {index} # ================================================")
    print(line)

======================================= #Line No 1 # ================================================
History and Physical Examination:

======================================= #Line No 2 # ================================================


======================================= #Line No 3 # ================================================
HISTORY:

======================================= #Line No 4 # ================================================
Presenting Complaints / Reasons: 

======================================= #Line No 5 # ================================================
 Tcell Lymphoblastic Lymphoma

======================================= #Line No 6 # ================================================
 Fever,Recurrent hypoglycaemia

======================================= #Line No 7 # ================================================
 Respiratory distress and loose stools

======================================= #Line No 8 # ================================================

In [12]:
with open('filterd_file.txt', 'r') as file:
    filterd_text_data =  file.read()

In [17]:
filterd_text_data

"History and Physical Examination:\n\nHISTORY:\nPresenting Complaints / Reasons: \n Tcell Lymphoblastic Lymphoma\n Fever,Recurrent hypoglycaemia\n Respiratory distress and loose stools\nHistory Of Present illness:\n T cell lymphoblastic lymphoma \n Induction day 16\n Fever came in emergency with complain of fever for\n 1 day without rigors and chills , associated with\n cough non productive .Has respiratory distress and\n recurrent hypoglycaemia\n Developed loose stools since morning multiple\n episodes as per attendant\n persistent Moaning and has weak peripheral\n pulses.CRT >2 seconds. \nPast Medical History: \n chemo related multiple admissions\nSocioeconomic History:\nMarital Status:\n minor\nFather's profession / Monthly income:\n dependant\nAny drugs of abuse/addictions:\n none\nOthers:\nFamily History: \n insignificant\nDevelopmental History:\n as per age\nVaccination History:\n as per epi\nREVIEW OF SYSTEMS: \nGeneral: \n Fever + , Malaise + , Fatigue +\nHeent: \n Sore throat 

###### Totel Length of filterd Text Data

In [13]:
len(filterd_text_data)

29104

###### Totel Length of filterd Text Data Courpus

In [14]:
filterd_text_data_courpus= filterd_text_data.split(" ")
len(filterd_text_data_courpus)

3817

# After filtering out 10434 words, we were left with 3817 words.

In [22]:
print(f"It's {len(input_text_data_courpus)-len(filterd_text_data_courpus)} words have been trimmed.")

It's 6617 words have been trimmed.
